### PR TITLE
feature: separating sagemaker dependencies into more use case specific installable components.

### PR DESCRIPTION
**Pull request by nadiaya**
Adding extras_require dependencies subsets for local mode and analytics dependencies. 

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -38,12 +38,35 @@ def read_version():
     "numpy>=1.9.0",
     "protobuf>=3.1",
     "scipy>=0.19.0",
-    "urllib3>=1.21, <1.25",
     "protobuf3-to-dict>=0.1.5",
     "requests>=2.20.0, <2.21",
     "fabric>=2.0",
```
Comment (laurenyu) [inline setup.py]: I realize this was part of an old change, but `fabric` should actually be part of test requires

File: setup.py [hunk 2]
```diff
@@ -38,12 +38,35 @@ def read_version():
     "numpy>=1.9.0",
     "protobuf>=3.1",
     "scipy>=0.19.0",
-    "urllib3>=1.21, <1.25",
     "protobuf3-to-dict>=0.1.5",
     "requests>=2.20.0, <2.21",
```
Comment (laurenyu) [inline setup.py]: I realize this isn't strictly in scope, but the request bounds were somewhat defined by docker-compose - we could probably change it to `requests>=2.20.0`?
* https://github.com/aws/sagemaker-python-sdk/pull/533
* https://github.com/aws/sagemaker-python-sdk/pull/538

File: setup.py [hunk 3]
```diff
@@ -38,12 +38,35 @@ def read_version():
...
     "fabric>=2.0",
 ]
 
+# Specific use case dependencies
+extras = {"analytics": ["pandas"], "local": ["urllib3>=1.21, <1.25", "docker-compose>=1.23.0"]}
```
Comment (laurenyu) [inline setup.py]: should we add one for `tensorflow`? (if people want to use the TF serializers/deserializers that require the library itself)

File: setup.py (hunk 1 above)
Comment (nadiaya) [reply setup.py]: Thanks!

File: setup.py (hunk 3 above)
Comment (nadiaya) [reply setup.py]: i didn't want to touch it, but actually why not :)

File: setup.py (hunk 2 above)
Comment (nadiaya) [reply setup.py]: Maybe, don't have bandwidth to do a proper check for it as part of this PR knowing how many problems it caused before.

Comment (laurenyu) [reply setup.py]: fair enough 😂

**Pull request merged**